### PR TITLE
Implement extract_data_from_fhir_search

### PR DESCRIPTION
**Pull request by spencer-skylight**
# Description
This PR adds a function that calls `extract_data_from_fhir_search_incremental` recursively until all paginated data has been retrieved from the FHIR server via search.

# Additional Changes
* Changed signature of `extract_data_from_fhir_search_incremental` to return `List[dict], str` rather than `List[Response], str`.
* Reordered some code to put internal/helper methods at the bottom of the file.

### CODE REVIEW

**Review by m-goggins — COMMENTED** [review]
Looks good to me

**Review by m-goggins — APPROVED** [review]

File: phdi/fhir/tabulation/tables.py [hunk 1]
```diff
@@ -222,6 +207,32 @@ def extract_data_from_fhir_search_incremental(
...
+
+    results, next = extract_data_from_fhir_search_incremental(
+        search_url=search_url, cred_manager=cred_manager
+    )
+
+    while next is not None:
+        incremental_results, next = extract_data_from_fhir_search_incremental(
+            search_url=next, cred_manager=cred_manager
+        )
+        results.extend(incremental_results)
```
Comment (bamader) [inline phdi/fhir/tabulation/tables.py:231]: More for my own information than a requested change, but once tabulate and write are done I'm assuming the plan will be to get rid of this `results.extend()` call in favor of incrementally processing and writing the page of data that we retrieved?

**Review by bamader — APPROVED** [review]
LGTM!

File: phdi/fhir/tabulation/tables.py (hunk 1 above)
Comment (spencer-skylight) [reply phdi/fhir/tabulation/tables.py:231]: This function will actually keep the iterative request and `.extend` call. There will be another function we add after the write logic is complete to do the incremental calls that you're describing.

**Pull request merged**